### PR TITLE
upgrade to React 0.14

### PR DESCRIPTION
**Pull request by conorhastings**
@davidtheclark

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
     "widget"
   ],
   "dependencies": {
-    "react": "0.13.x",
+    "react": "0.14.x",
+    "react-test-utils": "0.0.1",
```
Comment (davidtheclark) [inline package.json]: This (react-test-utils) should be a `devDependency`, right? Because the end-user doesn't need it?

File: package.json [hunk 2]
```diff
@@ -45,7 +45,8 @@
     "widget"
   ],
   "dependencies": {
-    "react": "0.13.x",
+    "react": "0.14.x",
```
Comment (davidtheclark) [inline package.json]: If we actually don't have to change anything about the plugin to make it compatible with 0.14, can we use a more flexible version range (cf. https://docs.npmjs.com/files/package.json#dependencies)? Couldn't the user have *either* 0.13.x *or* 0.14.x?

File: test/Menu-test.jsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import sinon from 'sinon';
 import React from 'react/addons';
 import Menu from '../src/Menu';
```
Comment (davidtheclark) [inline test/Menu-test.jsx]: If this is changing to an import I'd prefer to remove the empty line --- keep the imports all bunched together.

**Comment by davidtheclark** [comment]
Thanks for taking the initiative on this @conorhastings. I added some comments. Let me know what you think.

File: package.json (hunk 1 above)
Comment (conorhastings) [inline package.json]: updated this.

**Comment by conorhastings** [comment]
@davidtheclark unfortunately i had missed one piece of the upgrade , React.findDOMNode is now ReactDOM.findDOMNode so i think the dependency will have to remain unchanged. https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#changelog 

Let me know if there's anything else!

**Comment by davidtheclark** [comment]
Ah, breaking change -- too bad. Well, I'll have a look at this soon and probably cut a release. I might take a few days to try to think of whether there are other breaking changes I might want to incorporate into the same major release.

**Comment by conorhastings** [comment]
that sounds good, thanks for the feedback!

**Comment by conorhastings** [comment]
@davidtheclark i've also updated/rebuilt the demos now, let me know if you want me to leave that/or pull it out.

**Comment by davidtheclark** [comment]
You can pull it out. I'm going to take advantage of 0.14 and change the API a little to use `context` and simplify things, so I'll rewrite the demos accordingly.

**Comment by conorhastings** [comment]
cool, sounds good, done.

**Comment by davidtheclark** [comment]
@conorhastings : Released in 3.0.0. Thanks again.

**Pull request merged**